src/nvme/ioctl.c

```diff
@@ -126,23 +254,36 @@ bool nvme_get_debug(void)
 	return nvme_debug;
 }
 
+void nvme_set_latency(bool latency)
+{
+	nvme_latency = latency;
+}
+
+bool nvme_get_latency(void)
+{
+	return nvme_latency;
+}
+
 static int nvme_submit_passthru(int fd, unsigned long ioctl_cmd,
 				struct nvme_passthru_cmd *cmd, __u32 *result)
 {
 	struct timeval start;
 	struct timeval end;
 	int err;
 
-	if (nvme_get_debug())
+	if (nvme_get_latency())
 		gettimeofday(&start, NULL);
 
 	err = ioctl(fd, ioctl_cmd, cmd);
 
-	if (nvme_get_debug()) {
+	if (nvme_get_latency()) {
 		gettimeofday(&end, NULL);
-		nvme_show_command(cmd, err, start, end);
+		nvme_show_latency(ioctl_cmd == NVME_IOCTL_ADMIN_CMD, cmd->opcode, &start, &end);
 	}
 
+	if (nvme_get_debug())
+		nvme_show_command(cmd, err);
```

Comment:
Pre-existing, but curious why the 64-bit ioctls (`nvme_submit_passthru64()`) don't do this logging. Is it not useful for them? Does it just require more work to implement than it's worth?

Reply:
Yes will do implement as same for the 64-bit ioctls also.